extras/ping/ping.go

```diff
@@ -372,8 +376,9 @@ func (p *Pinger) runLoop(recvCh <-chan *packet) error {
 			return nil
 
 		case r := <-recvCh:
-			err := p.processPacket(r)
+			err := p.processPacket(r, &dstIP)
 			if err != nil {
+				fmt.Println("ERROR", err)
```

Comment:
Please, do not use `fmt.Println` to print errors. If there is a desire to log errors, we should have a customizable logger such that we can correctly redirect logs when running on mobile devices. Alternatively, if logging errors is useful when debugging and we do not care about bringing logs to users, I would recommend using the `log` package in the standard library, which defaults to printing on the standard error. We cannot print on the standard output because that is reserved for communication between `ooniprobe` and the Desktop app. Any message on the standard output that does not conform to the expected logging message would confuse the desktop app.

Reply:
understood, sorry about that. This one was probably a debugging slip; there were other places that were intended to print to console to mimick the `ping` utility, but this is a  library and any output function should live in the cmd entrypoint. fixed.